catlearn/ga/algorithm.py

```diff
@@ -399,9 +399,11 @@ def _cross_validate(args):
                                         axis=0)[:, parameter]
         train_targets = np.concatenate(train_targets, axis=0)
         try:
-            calc_fit += np.array(fit_func(
-                train_features, train_targets, test_features,
-                test_targets))
+            score = fit_func(train_features, train_targets,
+                             test_features, test_targets)
+            if len(score) != fitness_parameters:
+                raise AssertionError("len(fit_func) != fitness_parameters")
+            calc_fit += np.array(score)
         except ValueError:
```

Comment:
@pcjennings Do we really want to except ValueError? If we are looking only for the specific case of a singular matrix in the cholesky in the GP, we could instead import and except `LinAlgError`?